src/sqs/backfill.ts

```diff
@@ -1,17 +1,46 @@
 import {MessageHandler} from "./index"
-import {booleanFlag, BooleanFlags} from "../config/feature-flags";
-
+import app from "../worker/app";
+import {BackfillQueue, processInstallation} from "../sync/installation";
+import * as Sentry from "@sentry/node";
+import AxiosErrorEventDecorator from "../models/axios-error-event-decorator";
+import SentryScopeProxy from "../models/sentry-scope-proxy";
 
 export type BackfillMessagePayload = {
 	installationId: number,
 	jiraHost: string,
 	startTime?: string
 }
 
-export const backfillQueueMessageHandler : MessageHandler<BackfillMessagePayload> = async (context) => {
-	if(!await booleanFlag(BooleanFlags.NEW_BACKFILL_PROCESS_ENABLED, false, context.payload.jiraHost)) {
-		context.log.info("New backfill process disabled, dropping the message ");
-		return
+type BackfillQueueSupplier = () => Promise<BackfillQueue>;
+
+export const backfillQueueMessageHandlerFactory: (queueSupplier: BackfillQueueSupplier) => MessageHandler<BackfillMessagePayload> =
+	(queueSupplier) => async (context,) => {
```

Comment:
Copy-pasted from http://github.com/atlassian/github-for-jira/blob/d5d92be2f0e5d48e449e85b7ca0b8cd281c0b747/src/worker/startup.ts#L27-L27